api/v1alpha1/protectedapplicationview_types.go

```diff
@@ -0,0 +1,165 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ProtectedApplicationViewSpec defines the desired state of ProtectedApplicationView
+type ProtectedApplicationViewSpec struct {
+	// Type of application (ApplicationSet, Subscription, or Discovered)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=ApplicationSet;Subscription;Discovered
+	Type string `json:"type"`
+
+	// ApplicationRef references the source application resource
+	// +kubebuilder:validation:Required
+	ApplicationRef corev1.ObjectReference `json:"applicationRef"`
+
+	// Placements contains placement information for this application
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Placements []PlacementInfo `json:"placements,omitempty"`
+
+	// DRInfo contains disaster recovery configuration
+	// +kubebuilder:validation:Required
+	DRInfo DRInfo `json:"drInfo"`
+
+	// ApplicationSetInfo contains ApplicationSet-specific information
+	// +optional
+	ApplicationSetInfo *ApplicationSetInfo `json:"applicationSetInfo,omitempty"`
+
+	// SubscriptionInfo contains Subscription-specific information
+	// +optional
+	SubscriptionInfo *SubscriptionInfo `json:"subscriptionInfo,omitempty"`
+}
+
+// PlacementInfo contains information about application placement
+type PlacementInfo struct {
+	// Name of the Placement resource
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Namespace of the Placement resource
+	// +kubebuilder:validation:Required
+	Namespace string `json:"namespace"`
+
+	// Kind of placement (Placement or PlacementRule)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Placement;PlacementRule
+	Kind string `json:"kind"`
+
+	// SelectedClusters is the list of clusters where application is placed
+	// +optional
+	SelectedClusters []string `json:"selectedClusters,omitempty"`
+
+	// NumberOfClusters is the count of selected clusters
+	// +kubebuilder:validation:Minimum=0
+	NumberOfClusters int `json:"numberOfClusters"`
```

Comment:
Both of them provide _almost_ similar info, any added benefit of having extra fields in the API?
Any controller can simply do a `len()` of the "SelectedClusters" to compute and use this value in the code.